### PR TITLE
[Fix] delete cover image document update

### LINKED ISSUE #2173: [Bug] The system cannot find the file specified after deleting selected cover image
### Bug Description

After deleting the selected image of a document cover will cause the following exception:
 (OS Error: The system cannot find the file specified, errno = 2)

### How to Reproduce

- Add a document cover image
- delete that image through the change cover popup
- restart the app

### Expected Behavior

It should revert back to default no cover image

### Operating System

Windows 10

### AppFlowy Version(s)

0.1.2

### Screenshots

image

image

### Additional Context

https://github.com/AppFlowy-IO/AppFlowy/pull/2111#issuecomment-1493578791

### PR DESCRIPTION
**Pull request by GouravShDev**
fixes #2173 

- Added alert dialog before deleting current document's cover image
- Updates current document state on cover delete by user
- Reset document cover state to "No Cover" if file not found while building widget

I am currently only checking for current document cover image when user deletes any image. I'm not sure how I can check if image is used by any other document.

https://user-images.githubusercontent.com/74348508/230787506-4b240bb2-2405-4c1e-b9d6-77b3b510acd6.mp4

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart [hunk 1]
```diff
@@ -435,8 +435,18 @@ class _CoverImageState extends State<_CoverImage> {
...
+        final imageFile =
+            File(widget.node.attributes[kCoverSelectionAttribute]);
+        if (!imageFile.existsSync()) {
+          // reset cover state
+          WidgetsBinding.instance.addPostFrameCallback((_) {
+            widget.onCoverChanged(CoverSelectionType.initial, null);
+          });
+          coverImage = const SizedBox();
+          break;
+        }
```
Comment (GouravShDev) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart:447]: @LucasXu0.  Here I added backup logic if file was not found reset to no cover state

Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart:447]: Got it.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/change_cover_popover.dart [hunk 2]
```diff
@@ -279,14 +322,66 @@ class _ChangeCoverPopoverState extends State<ChangeCoverPopover> {
...
+      ),
+      content: Container(
+        constraints: const BoxConstraints(minHeight: 100),
+        padding: const EdgeInsets.symmetric(
+          vertical: 20,
+        ),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: const [
+            Text("It will be removed from cover after it is deleted."),
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/change_cover_popover.dart:349]: Please add this text to the translation file.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/change_cover_popover.dart [hunk 3]
```diff
@@ -279,14 +322,66 @@ class _ChangeCoverPopoverState extends State<ChangeCoverPopover> {
...
+            SizedBox(
+              height: 4,
+            ),
+            Text("Are you sure, you want to continue?"),
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/change_cover_popover.dart:353]: Same as the previous.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/change_cover_popover.dart (hunk 2 above)
Comment (GouravShDev) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/change_cover_popover.dart:349]: ok

**Comment by LucasXu0** [comment]
@GouravShDev I added `require_trailing_commas` rules in `analysis_options.yaml` file. Please merge or rebase the main branch and run the `dart format .` in `frontend/appflowy_flutter` to format your code.

**Comment by annieappflowy** [comment]
This PR needs to get merged into v0.1.3

**Comment by GouravShDev** [comment]
Are there changes I need to do for this PR ?

**Comment by LucasXu0** [comment]
LGTM.

**Pull request merged**